server/viame_server/viame_detection.py

```diff
@@ -234,19 +247,9 @@ def stream():
         )
     )
     def get_detection(self, folder):
-        detectionItems = list(
-            Item().findWithPermissions(
-                {"meta.detection": str(folder["_id"])}, user=self.getCurrentUser(),
-            )
-        )
-        detectionItems.sort(key=lambda d: d["created"], reverse=True)
-        if not len(detectionItems):
-            return None
-        file = Item().childFiles(detectionItems[0])[0]
-
-        # TODO: deprecated, remove after we migrate to json
+        file = self._load_detections(folder)
         if "csv" in file["exts"]:
-            return viame.load_csv_as_tracks(file)
+            return getTrackData(file)
```

Comment:
This function could theoretically be a single line: `return getTrackData(self._load_detections(folder))`

However, in the expected case, this would cause json tracks to buffer into memory and get parsed and re-serialized.  Avoid this by streaming them.